### ISSUE TITLE
Kubernetes extension generates environment variables twice

### ISSUE THREAD

**Issue by Ladicek** [quarkusio/quarkus#7740]
**Describe the bug**

I use the Kubernetes extension configuration properties to add environment variables to the `Deployment` / `DeploymentConfig`, and I'm getting the same piece of YAML generated twice.

This happens at least for generating env var from a ConfigMap and from a secret, both in the "one env var for specific key" and "env vars for all keys" scenarios.

For example, I have this config:

```properties
# one env var for specific key in the secret
quarkus.openshift.env-vars.MY_ENV_VAR.secret=the-secret
quarkus.openshift.env-vars.MY_ENV_VAR.value=key-in-the-secret

# env vars for all keys in the ConfigMap
quarkus.openshift.env-vars.IGNORED.configmap=the-configmap
```

With this, the generated YAML contains:

```yaml
    spec:
      containers:
      - env:
        - name: "MY_ENV_VAR"
          valueFrom:
            secretKeyRef:
              key: "key-in-the-secret"
              name: "the-secret"
              optional: false
        - name: "MY_ENV_VAR"
          valueFrom:
            secretKeyRef:
              key: "key-in-the-secret"
              name: "the-secret"
              optional: false
        envFrom:
        - configMapRef:
            name: "the-configmap"
            optional: false
        - configMapRef:
            name: "the-configmap"
            optional: false
```

In the debugger, I see `AddEnvVarDecorator.andThenVisit` called twice during `session.close()`, but as far as I was able to observe, the `AddEnvVarDecorator` is only added _once_ in the Quarkus extension.

**Expected behavior**
Only one generated piece of YAML for each config key.

**Actual behavior**
Two identical pieces of YAML for each config key.

**To Reproduce**
1. Add the `quarkus-kubernetes` extension.
2. Run `mvn clean package -DskipTests`.
3. See `target/kubernetes/*.yml`.

**Configuration**
See above.

**Comment by Ladicek** [comment]
CC @geoand @iocanel

**Comment by geoand** [comment]
@iocanel a dekorate bug perhaps?

**Comment by iocanel** [comment]
Possibly, I'll have a look.

**Comment by Ladicek** [comment]
This was _partially_ fixed by #7749.

**Comment by Ladicek** [comment]
Clicked the wrong button :-)
